frappe/email/receive.py

```diff
@@ -357,9 +357,13 @@ def __init__(self, content):
 		"""Parses headers, content, attachments from given raw message.
 
 		:param content: Raw message."""
-		self.raw = safe_encode(content) if six.PY2 else safe_decode(content)
-		self.mail = email.message_from_string(self.raw)
-
+		if six.PY2:
+			self.mail = email.message_from_string( safe_encode(content) )
+		else:
+			if isinstance(content, bytes):
+				self.mail = email.message_from_bytes(content)
+			else:
+				self.mail = email.message_from_string( safe_decode(content) )
```

Comment:
here as well

Reply:
Jus so you know, `self.mail = email.message_from_string( safe_decode(content) )` is redundant piece of code. Correct me if I'm wrong but `content` is either `bytes` or `str` which means that `email.message_from_bytes(content) == email.message_from_string( safe_decode(content) )` because is content is str `safe_decode` will return same str.
My suggestion is ensuring that Email.__init__ always gets content in form of bytes and getting rid of this redundant line.

Reply:
yes, that's right. you can probably get rid of `safe_decode()` and it'll still work just fine.